File: JavaSnippets/src/animals/Dog.java
package animals;

public class Dog extends Animal {
	
	// Default Constructor
	public Dog() {
		name = "";
		age = 0;
	}
	
	// Constructor with parameters
	public Dog(int age, String name) {
		this.age = age;
		this.name = name;
	}
	
	public void bark() {
		System.out.println("Wuff");
	}
}

File: CppSnippets/animals/Dog.cpp
/*
 * Dog.cpp
 *
 *  Created on: 12.03.2018
 *      Author: markus
 */

#include "Dog.h"
#include <iostream>


// Default Constructor
Dog::Dog() {
	this->age = 0;
	this->name = "";
}

// Constructor with parameters
Dog::Dog(int age, std::string name) {
	this->age = age;
	this->name = name;
}


Dog::~Dog() {
	std::cout << this->name << " left" << std::endl;
}

